### PR TITLE
Add AdaBoost model and option to train on a subset of features

### PR DESCRIPTION
**Pull request by luccalb**
Added the following:

- AdaBoost Classifier (DecisionTree under the hood)
- Option to select a subset of features for training in EVP demo
- Add results to documentation

### CODE REVIEW

**Review by ultiwinter — COMMENTED** [review]
I didn't run it yet, but looks good. You could as well add in the documentation its performance.

**Comment by luccalb** [comment]
> I didn't run it yet, but looks good. You could as well add in the documentation its performance.

Just added the docs :)

File: src/evp/predictors.py [hunk 1]
```diff
@@ -55,7 +56,7 @@ def predict(self, X) -> list[MerchantSizeByDPV]:
     def train(
         self, X_train, y_train, X_test, y_test, epochs=1, batch_size=None
     ) -> None:
-        log.info(f"Training {type(self).__name__}")
+        log.info(f"Training {type(self).__name__} for {epochs} epochs")
```
Comment (ultiwinter) [inline src/evp/predictors.py:59]: Shouldn't the conventional epochs for simpler ML to be always 1? I am not familiar with AdaBoost architecture

**Review by ultiwinter — APPROVED** [review]
Works well. Just a comment/question.

File: src/evp/predictors.py (hunk 1 above)
Comment (luccalb) [reply src/evp/predictors.py:59]: Yes, you're right

**Pull request merged**